#!/usr/bin/env python
PACKAGE = "velodyne_detection"
NODE_NAME = "velodyne_detection"
PARAMS_NAME = "DetectionNode"

from math import pi
import dynamic_reconfigure.parameter_generator_catkin as pgc

gen = pgc.ParameterGenerator()

gen.add("angle_offset",
  pgc.double_t,
  0,
  "angle_offset",
  0.0, -pi, pi)

gen.add("track_width",
  pgc.double_t,
  0,
  "track_width",
  1.524, 0.1, 3.0)

gen.add("safe_pad",
  pgc.double_t,
  0,
  "safe_pad",
  0.1524, 0.0, 1.0)

gen.add("ClusterTolerance",
  pgc.double_t,
  0,
  "ClusterTolerance",
  0.3, 0.0, 999.999)

gen.add("MinClusterSize",
  pgc.int_t,
  0,
  "MinClusterSize",
  10, 0, 99999)

gen.add("MaxClusterSize",
  pgc.int_t,
  0,
  "MaxClusterSize",
  600, 0, 99999)

exit(gen.generate(PACKAGE, NODE_NAME, PARAMS_NAME))
